dev_docs/tutorials/versioning_interfaces.mdx

```diff
@@ -0,0 +1,418 @@
+}
+
+type UnsavedWeatherDashboard = Omit<WeatherDashboard, 'id'>;
+
+/** Below are the HTTP APIs based on the domain types */
+
+interface GetWeatherDashboardHTTPResponse {
+  weatherDashboard: WeatherDashboard;
+}
+
+interface CreateWeatherDashboardHTTPBody {
+  weatherDashboard: UnsavedWeatherDashboard;
+}
+
+interface CreateWeatherDashboardHTTPResponse {
+  weatherDashboard: WeatherDashboard;
+}
+
+// Imagine this pattern for the update and list endpoints too.
+```
+
+### An example: Weather insights versioned
+
+The following product requirements are added to the Weather insights plugin:
+
+1. Data sources need a more user-friendly name, currently we just show a URL
+2. New feature: **Predictions** that will generate forecasts based on past and current weather data
+
+Before we make these changes, we need to make sure that our current interfaces are not lost.
+
+#### Preparing our interfaces for versions
+
+We are going to make our current set of interfaces `v1`. Let's reflect this in our code by creating this file structure in `common`:
+
+```
+common
+  index.ts
+  latest.ts
+  data_source
+    v1.ts
+  weather_dashboard
+    v1.ts
+public
+  ...
+server
+  ...
+```
+
+By placing `DataSource` and `WeatherDashboard` in two folders we indicate that they _could_ version independently. Specifically,
+`WeatherDashboard` may version independently of `DataSource`.
+
+**Note:** At this stage, splitting these concepts may be overkill if we expect `WeatherDashboard` to _rarely_ change, but for this example we would like the ability to version them separately.
+
+The `latest.ts` file is going to act as an alias to the latest set of interfaces. It will look like this:
+
+```ts
+// "export *" is considered safe here because we ONLY have types in these files.
+export * from './data_source/v1';
+export * from './weather_data/v1';
+```
+
+Our `index.ts` file will contain:
+
+```ts
+// Explicit export of everything from latest
+export { DataSource, WeatherDashboard, GetWeatherDashboardHTTPResponse } from './latest';
+
+export * as weatherDashboardV1 from './weather_dashboard/v1';
+export * as dataSourceV1 from './data_source/v1';
+```
+
+How we use these types depends on whether the code is aware of versions or not. Generally, code that is aware of multiple versions will create greater complexity.
+
+```ts
+// This type is being pulled from "latest", so whenever a new latest is linked,
+// it will also point to the latest WeatherDashboard.
+import type { WeatherDashboard } from '../common';
+
+function logIt(dashboard: WeatherDashboard) {
+  logger.debug(dashboard);
+}
+```
+
+<DocCallOut title="Minimize the impact of refactors">
+  By referencing types linked from `latest.ts` we will not have to refactor all our app code to point to the latest version of `WeatherDashboard`. This happens by simply re-exporting the newest types from `latest.ts`.
+</DocCallOut>
+
+```ts
+// Consider this route handler that expects V1 weather dashboard requests
```

Comment:
It feels like having a similar example early on in the document would be helpful. It makes the abstract "interfaces" a bit more concrete to show that we're creating types describing the shape of data coming into and out of the HTTP API. 

So maybe after describing the unversioned types we can have an example unversioned handler.

Then here maybe if the comment could be more verbose to say what this example shows. Something like "Putting it all together we now have the types to describe the request and response payloads of the different versions of our API". Perhaps a create example could be a better illustration as we could have the request payload typed and the response payload.